File: 0049 group-anagrams/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
    Map<Integer,List<String>> map = new HashMap<>();
    for(String s:strs) {
        int key = getKey(s);
        if(!map.containsKey(key)) {
            map.put(key,new ArrayList<>());
        }
        map.get(key).add(s); 
    }
    return new ArrayList<>(map.values());
    }
    private int getKey(String s) {
        int[] d=new int[26];
        for(char c:s.toCharArray()) d[c-'a']++;
        return Arrays.hashCode(d);
    }
}

File: 0049 group-anagrams/solution.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string,List<string>> map =new Dictionary<string, List<string>>();
        foreach(var s in strs) {
            var count = new char[26];
            foreach(char c in s) count[c-'a']++;
            string key = new string(count);
            if(!map.ContainsKey(key)) {
                map[key]=new List<string>();
            }
            map[key].Add(s);
        }
        return new List<IList<string>>(map.Values);
    }
}
